CounterStrike/src/dev/danablend/counterstrike/CounterStrike.java

```diff
@@ -1006,7 +1112,8 @@ public void returnPlayertoGame(CSPlayer csplay) {
 
         Player player = csplay.getPlayer();
 
-        if (ResourseHash.get(player.getName() + "RES") == null || ResourseHash.get(player.getName() + "RES") == "DEFAULT") {
+        //isOnline filters NPCs
+        if (player.isOnline() && (ResourseHash.get(player.getName() + "RES") == null || ResourseHash.get(player.getName() + "RES") == "DEFAULT")) {

```

Comment:
Same as before, use .equals().
Btw, for readability you could extract `ResourseHash.get(player.getName() + "RES"` to a variable - in this method it is used 4 times.